waf_test.go

```
@@ -246,3 +252,210 @@ func TestRoundRobinEmptyPool(t *testing.T) {
+	var calls atomic.Int32
+	e1 := &Engine{addr: "192.0.2.1:8000", maxFails: 0, detectFn: func(*http.Request) (*detection.Result, error) {
+		calls.Add(1)
+		return nil, errors.New("connection refused")
+	}}
+	e2 := &Engine{addr: "192.0.2.2:8000", maxFails: 0, detectFn: func(*http.Request) (*detection.Result, error) {
+		calls.Add(1)
+		return &detection.Result{Head: '.'}, nil
+	}}
+
+	m := newTestWAF(EnginePool{e1, e2}, 0)
+	req := httptest.NewRequest(http.MethodGet, "http://example.com/", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)